app/views/images/index.html.erb

```diff
@@ -3,35 +3,37 @@
 
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-two-thirds">
+    <div class="app-pane">
+      <% upload_image_heading = capture do %>
+        <h2 class="govuk-heading-m">
+          <%= t("images.index.upload_image") %>
+        </h2>
+      <% end %>
 
-    <% unless rendering_context == 'modal' %>
-      <div class="app-pane">
-        <% upload_image_heading = capture do %>
-          <h2 class="govuk-heading-m">
-            <%= t("images.index.upload_image") %>
-          </h2>
-        <% end %>
-
-        <%= form_tag(
-          create_image_path(@document),
-          multipart: true,
-          data: { gtm: "image-upload-submit" },
-        ) do %>
-          <%= render "govuk_publishing_components/components/file_upload", {
-            label: {
-              text: upload_image_heading,
-            },
-            name: "image"
-          } %>
-
-          <%= render_govspeak(t("images.index.description_govspeak")) %>
-
-          <%= render "govuk_publishing_components/components/button", {
-            text: "Upload",
-          } %>
-        <% end %>
-      </div>
-    <% end %>
+      <%= form_tag(
+        create_image_path(@document),
+        id: "image-upload",
+        multipart: true,
+        data: { gtm: "image-upload-submit" },
+      ) do %>
+        <%= render "govuk_publishing_components/components/file_upload", {
+          label: {
+            text: upload_image_heading,
+          },
+          name: "image"
+        } %>
+
+        <%= render_govspeak(t("images.index.description_govspeak")) %>
+
+        <%= render "govuk_publishing_components/components/button", {
+          text: "Upload",
+          data_attributes: {
+            "modal-action": "upload",
+            "modal-action-form": "image-upload",
```

Comment:
As we put the action rules on this button are we vulnerable to this not working if someone submits this form without clicking the button.

I'm not sure if we can do it on this form as it only has the input file which is a very browser specific action, but once we have the edit form and pressing enter can submit the form these events wouldn't be caught.

Reply:
Ooh, good thought - that hadn't occurred to me. To my surprise, it works out-of-the-box, thanks to the spec: https://www.w3.org/TR/html52/sec-forms.html#implicit-submission. I realise that's HTML, but it seems to work in IE11 as well, and I'm just checking other browsers.

Reply:
Oh cool - I had no idea about the implicit submission.